src/main/scala/me/chuwy/otusfp/hw/services/ApiService.scala

```diff
@@ -0,0 +1,70 @@
+package me.chuwy.otusfp.hw.services
+
+import cats.effect.kernel.Async
+import cats.effect.std.Dispatcher
+import cats.effect.{Concurrent, IO, Ref, Sync, Temporal}
+import fs2.Stream
+import io.circe.{Encoder, Json}
+import org.http4s.EntityEncoder
+import org.http4s.circe.jsonEncoderOf
+import cats.implicits._
+
+import scala.concurrent.duration.DurationInt
+
+object ApiService {
+
+  trait Api[F[_]] {
+    def counter(): F[Api.Counter]
+
+    def getStream(chunk: Int, total: Int, time: Int): Stream[F, String]
+  }
+
+  object Api {
+    final case class Counter(count: Int)
+
+    final case class CounterEnv[F[_]](count: Ref[F, Int]) {
+      def increment: F[Int] = this.count.updateAndGet(_ + 1)
+    }
+
+    object CounterEnv {
+      def build[F[_] : Sync](): F[CounterEnv[F]] = for {
+        ref <- Ref[F].of(0)
+      } yield CounterEnv(ref)
```

Comment:
```suggestion
      def build[F[_] : Sync]: F[CounterEnv[F]] =
        Ref[F].of(0).map(ref => CounterEnv(ref))
```

Лучше избегать одно-строчных for-comprehensions.